lib/iob/calculate.js

```diff
@@ -37,4 +58,76 @@ function iobCalc(treatment, time, dia) {
     return results;
 }
 
-exports = module.exports = iobCalc;
+
+function iobCalc(treatment, time, dia, profile) {
+
+	var curve = profile.curve;
+
+    if (curve === undefined || curve.toLowerCase() == 'bilinear') {
+    	return iobCalcBiLinear(treatment, time, dia);
+    };
+    
+    curve = curve.toLowerCase();
+    
+    if (curve != 'novorapid' && curve != 'fiasp') {
+    	console.error('Unsupported curve function: "' + curve + '". Supported curves: bilinear, novorapid and fiasp.');
+    	return;
+    }
+
+	var usePeakTime = false;
+
+	if (profile.dia !== 5 && profile.insulinPeakTime !== undefined) {
+		console.error('Pump DIA must be set to 5 hours if using the Insulin Peak Time setting');
+		
+		if (profile.insulinPeakTime < 35 || profile.insulinPeakTime > 80) {
+			console.error('Insulin Peak Time is only supported for values between 35 to 80 minutes');
+		}
+		return;
+	}
+	
+	if (profile.insulinPeakTime !== undefined) {
+		usePeakTime = true;
+	}
+
+	// default to the published Fiasp DIA
+
+    var diaratio =  5.0 / profile.dia;
+
+	if (profile.insulinPeakTime !== undefined) {
+		diaratio = 55.0 / Number(profile.insulinPeakTime);
+	}
+    
+    var end = 300;
+
+    if (typeof time === 'undefined') {
+        time = new Date();
+    }
+
+    var results = {};
+
+    if (treatment.insulin) {
+        var bolusTime = new Date(treatment.date);
+        var minAgo = diaratio * (time - bolusTime) / 1000 / 60;
+        var iobContrib = 0;
+        var activityContrib = 0;
+
+        if (minAgo < end) {
+            if (curve.toLowerCase() === 'fiasp') {
+                iobContrib = treatment.insulin * (((minAgo / 55) + 1) * Math.exp(-(minAgo) / 55));
```

Comment:
Yeah the AUC changes; here's what the AUC normalized curves look like. The change in the calculated activity is pretty dramatic, wonder if this would work.

![auc normalized](https://user-images.githubusercontent.com/1001330/28584211-90b6c81c-7174-11e7-818a-9ff4e622e383.png)